File: RedSheet.DbEntities/Models/CulturePlan.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("CulturePlans", Schema = "dbo")]
	[Serializable]
	public partial class CulturePlan  
	{
 
	[Range(1, int.MaxValue)]
	public int CultureCountryId { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int CulturePlanId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string CulturePlanValue { get; set; } 
 
	[RelationshipTableAttribue("ApplicationObjects","dbo")]
	public CulturePlanCategory CulturePlanCategoryId { get; set; } 
	[InverseProperty("CulturePlan")]
	public ICollection<ProjectCulturePlan> ProjectCulturePlans { get; set; } 


	public CulturePlan()
	{
	this.ProjectCulturePlans = new HashSet<ProjectCulturePlan>();

    }

    }
}

File: redsheet-portal/src/app/database-models/culture-plan.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ProjectCulturePlan,  } from './'
export class CulturePlan {
    constructor(culturePlan?: CulturePlan )  {
        let properties = [ "cultureCountryId", "culturePlanId", "culturePlanValue", "culturePlanCategoryId", "projectCulturePlans",];
        for (let property of properties)
            if (culturePlan && culturePlan[property])
                this[property] = culturePlan[property];
    }
 
    @range(1,2147483647)
	cultureCountryId : number =   undefined;
 
	culturePlanId : number =   0 ;
 
    @required()
    @maxLength(1000)
	culturePlanValue : string =   undefined;
 
    @range(0,2147483647)
	culturePlanCategoryId : number =   undefined;
	@nested(ProjectCulturePlan)
	projectCulturePlans: ProjectCulturePlan[];



}
